src/backend/features/remote-sync/file-explorer/sync-remote-changes-to-local.test.ts

```diff
@@ -1,91 +1,71 @@
 import { syncRemoteChangesToLocal } from './sync-remote-changes-to-local';
-import { deepMocked, mockProps, partialSpyOn } from '@/tests/vitest/utils.helper.test';
+import { call, calls, mockProps, partialSpyOn } from '@/tests/vitest/utils.helper.test';
 import { AbsolutePath } from '@/context/local/localFile/infrastructure/AbsolutePath';
 import { FileUuid } from '@/apps/main/database/entities/DriveFile';
-import { unlink } from 'node:fs/promises';
 import { loggerMock } from '@/tests/vitest/mocks.helper.test';
-import { existsSync } from 'node:fs';
 import { Addon } from '@/node-win/addon-wrapper';
 
 vi.mock(import('node:fs/promises'));
 vi.mock(import('node:fs'));
 
 describe('sync-remote-to-local', () => {
-  const createFilePlaceholderMock = partialSpyOn(Addon, 'createFilePlaceholder');
+  partialSpyOn(Addon, 'setPinState');
+  const updatePlaceholderMock = partialSpyOn(Addon, 'updatePlaceholder');
 
-  const unlinkMock = deepMocked(unlink);
-  const existsSyncMock = deepMocked(existsSync);
-
-  const date = '2025-01-01T00:00:00.000Z';
-  const time = new Date(date).getTime();
+  const localDate = new Date('1999-01-01T00:00:00.000Z');
+  const remoteDate = new Date('2025-01-01T00:00:00.000Z');
   let props: Parameters<typeof syncRemoteChangesToLocal>[0];
 
   beforeEach(() => {
-    existsSyncMock.mockReturnValue(true);
     props = mockProps<typeof syncRemoteChangesToLocal>({
       local: {
         path: 'localPath' as AbsolutePath,
         stats: {
           size: 512,
-          mtime: new Date('1999-01-01T00:00:00.000Z'),
+          mtime: localDate,
         },
       },
       remote: {
         absolutePath: 'remotePath' as AbsolutePath,
         uuid: 'uuid' as FileUuid,
         name: 'file2',
         size: 1024,
-        createdAt: date,
-        updatedAt: date,
+        updatedAt: remoteDate.toISOString(),
       },
     });
   });
 
-  it('should sync remote changes to local when remote file is newer and has different size', async () => {
-    // Given/When
+  it('should not sync when remote file is older', async () => {
+    // Given
+    props.local.stats.mtime = new Date('2026-01-01T00:00:00.000Z');
+    // When
     await syncRemoteChangesToLocal(props);
     // Then
-    expect(unlinkMock).toBeCalledWith('localPath');
-    expect(createFilePlaceholderMock).toBeCalledWith({
-      path: 'remotePath',
-      placeholderId: 'FILE:uuid',
-      size: 1024,
-      creationTime: time,
-      lastWriteTime: time,
-    });
+    calls(updatePlaceholderMock).toHaveLength(0);
   });
 
-  it('should not sync when remote file is not newer', async () => {
+  it('should not sync when file sizes are equal', async () => {
```

Comment:
You should check at least the updated_at file also. Same sized-files with diff content will be accepted, otherwise. Or does this only apply to remote-only placeholders?

Reply:
> Currently the best we can do is use size and check that remote updatedAt is newer than local modification time (which we already do).

Yeah, this is what I am talking about. It should be done. But together. The legend of this test suggests that only checking the size is enough